c++11/test/in_memory_recorder.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <mutex>
+#include <stdexcept>
+#include <vector>
+#include "../src/recorder.h"
+
+namespace lightstep {
+// InMemoryRecorder is used for testing only.
+class InMemoryRecorder : public Recorder {
+ public:
+  void RecordSpan(collector::Span&& span) noexcept override {
+    std::lock_guard<std::mutex> lock_guard{mutex_};
+    spans_.emplace_back(std::move(span));
+  }
+
+  std::vector<collector::Span> spans() const {
+    std::lock_guard<std::mutex> lock_guard{mutex_};
+    return spans_;
+  }
+
+  size_t size() const {
```

Comment:
`size()` + guard doesn't play well. By the time you read the value it's something else => wrong API (if used beyond logging).